### DISCUSSION TITLE
Execution plan not working in Redash

### DISCUSSION THREAD

**Discussion by lscapim** [getredash/redash#6472] [General]
Hi, how do we use an execution plan in Redash?
When we try, we get this error:
![image](https://github.com/getredash/redash/assets/96078362/0c370831-2ba1-49fa-9cc8-3d35194916d2)

**Comment by lscapim** [comment]
Can someone help me?

**Reply by lscapim** [reply]
Hello,
How could I make this change?
https://discuss.redash.io/t/execution-plan-in-redash/10152
![image](https://github.com/getredash/redash/assets/96078362/85a55b91-e5f5-42d1-9d92-8fa94f375b99)

**Reply by guidopetri** [reply]
@lscapim you'd probably have to modify the base query runner defined [here](https://github.com/getredash/redash/blob/master/redash/query_runner/__init__.py#L117). Do note that (if I'm not mistaken) not all data sources will support plan explanations.

However, as mentioned by Jesse in the Discuss topic, Redash isn't meant to be a full SQL engine, but rather a visualization tool - as a result, I don't know how useful the plan explanation would be in a visualization. My suggestion would be to instead run this sort of plan explanation externally to Redash, and copy over the query once you're satisfied with the execution plan.

**Reply by guidopetri** [reply]
Additionally - bear in mind that a PR like that won't necessarily be accepted, due to it being such a big departure from the visualization focus of Redash. Feel free to join the [monthly meeting happening tomorrow](https://github.com/getredash/redash/wiki/Monthly-meeting-notes) if you'd like to discuss further with the rest of the maintainers!

**Comment by justinclift** [comment]
Being able to run explain statements from the Redash query page directly, instead of needing to manually log into the data source to do it, would probably make things easier for our users.

No idea how hard it would be to implement such a thing though.
